File: JavaScript/Algorithms/Recursion/factorial.js
/**
 * Calculates the factorial of a number using recursion
 *
 * @param {int} number
 * @return int
 */
function factorialRecursion(number) {
    if (number == 0) {
        return 1;
    }

    return number * factorialRecursion(number - 1);
}

File: C++/Algorithms/Math/factorial.cpp
/**
 * Calculates the factorial of a number using iteration
 *
 * @param int number
 * @return int
 */
int factorialIteration(int number) {
    int factorial = 1;

    for (int i = 2;i < number + 1;i++) {
        factorial *= i;
    }

    return factorial;
}